### PR TITLE
[Bugfix][Quantization] Support packed parameters with HSDP

### PR DESCRIPTION
**Pull request by pst2154**
## Purpose

Low-bit checkpoints can represent packed weights as integer `Parameter` tensors and global scales as scalar `Parameter` tensors. PyTorch FSDP2 cannot shard non-floating or zero-dimensional parameters. In addition, quantization post-load transforms must run while parameters are local tensors; running them after HSDP wrapping can invoke unsupported operations on DTensors.

Before this fix, loading a ModelOpt NVFP4 diffusion checkpoint with HSDP could fail while wrapping packed/scalar parameters or while post-processing quantized DTensors.

Reproduction before the fix:

```bash
vllm-omni serve <modelopt-nvfp4-diffusion-checkpoint> \
  --omni --enforce-eager --no-guardrails \
  --use-hsdp --hsdp-shard-size 2
```

This change:

- runs quantization post-load processing before HSDP wraps parameters as DTensors;
- detects packed integer and scalar parameters that FSDP2 cannot shard;
- moves those parameters to the target rank device and keeps them replicated;
- passes each nested `fully_shard` call only the ignored parameters owned by that module;
- adds regression coverage that uses a real one-rank Gloo process group and CPU `DeviceMesh`, proving float parameters become DTensors while packed/scalar parameters remain local.

This PR is independent and based directly on `main`.

## Test Plan

**vLLM Version:** 0.25.0

**vLLM-Omni Commit:** `d4da4fdd`

```bash
PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 python -m pytest -q \
  -p pytest_asyncio.plugin -p pytest_mock \
  tests/diffusion/model_loader/test_modelopt_fp8_adapter.py \
  tests/diffusion/distributed/test_hsdp.py
```

Two-GPU hardware test:

```bash
vllm-omni serve <modelopt-nvfp4-cosmos3-checkpoint> \
  --omni --enforce-eager --no-guardrails \
  --use-hsdp --hsdp-shard-size 2
```

Then submit a 256x256, 2-step request to `/v1/images/generations`.

## Test Result

- Current reviewer-focused suite: **33 passed, 16 warnings** in the pinned vLLM 0.25.0 container, including the real FSDP2 regression.
- Earlier five-file focused suite: **84 passed, 22 warnings** on vLLM 0.25.0.
- 2x NVIDIA B300 SXM6 AC, 275040 MiB each, compute capability 10.3.
- Both ranks initialized with `replicate_size=1`, `shard_size=2`, `world_size=2`.
- Each rank excluded the expected 2520 unshardable tensors: 504 non-floating packed tensors and 2016 scalar tensors, plus 4 model-declared ignored tensors.
- HSDP wrapped **72 modules + root** on both ranks with `ignored_params=2524`.
- Model loaded successfully at approximately 10.67 GiB per rank.
- End-to-end request returned HTTP 200 and a valid 256x256 PNG; steady pipeline time was 0.59 seconds after warmup.
- Hardware integration used the open #5076 ModelOpt/Cosmos3 prerequisite plus the NVFP4 scale-remap follow-up to provide a loadable unified checkpoint; this HSDP branch itself remains based directly on `main`.
- `ruff check`, `ruff format --check`, and `git diff --check` pass.

The commit is signed off and the PR is ready for review.

### CODE REVIEW

File: tests/diffusion/distributed/test_hsdp.py [hunk 1]
```diff
@@ -6,14 +6,54 @@
...
+        self.root_weight = nn.Parameter(torch.ones(2))
+
+
+def test_hsdp_ignores_packed_and_scalar_parameters_in_nested_wraps(mocker):
+    model = _PackedModel()
+    ignored_params = _unshardable_parameters(model)
+    expected_ignored = {model.block.packed_weight, model.block.input_global_scale}
+    assert ignored_params == expected_ignored
+
+    fully_shard = mocker.patch("vllm_omni.diffusion.distributed.hsdp.fully_shard")
```
Comment (david6666666) [inline tests/diffusion/distributed/test_hsdp.py]: This only checks the kwargs passed to a mock, so it won't catch the FSDP2 failure this bug fixes. Can we exercise a real `fully_shard` call here and assert that packed/scalar params stay local while float params become DTensors? A tiny CPU process group should be enough.

Comment (pst2154) [reply tests/diffusion/distributed/test_hsdp.py]: Agreed. Commit `d4da4fdd` replaces the mock-only assertion with a real one-rank Gloo process group and CPU `DeviceMesh`, then calls the actual FSDP2 `fully_shard` path through `shard_model`. The test asserts both float parameters become `DTensor`s while the packed uint8 and scalar parameters retain their original local identities. The reviewer-focused suite passes: 33 tests on vLLM 0.25.0.

Comment (david6666666) [reply tests/diffusion/distributed/test_hsdp.py]: Looks good now, thanks.

**Comment by david6666666** [comment]
LGTM now. Thank you for your contribution.

**Review by david6666666 — APPROVED** [review]
